### ISSUE TITLE
Compilation Error on MacOS: invalid subtyping in definition of _Clock

### ISSUE THREAD

**Issue by Clemapfel** [Clemapfel/mousetrap.jl#4]
Compilation of the Julia component throws (on aarch64, MacOS)

```julia
C++ exception while wrapping module detail: invalid subtyping in definition of _Clock with supertype Any
ERROR: LoadError: invalid subtyping in definition of _Clock with supertype Any
Stacktrace:
 [1] register_julia_module
   @ ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:393 [inlined]
 [2] readmodule(so_path::String, funcname::Symbol, m::Module, flags::Nothing)
   @ CxxWrap.CxxWrapCore ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:742
 [3] wrapmodule(so_path::String, funcname::Symbol, m::Module, flags::Nothing)
   @ CxxWrap.CxxWrapCore ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:747
 [4] include
   @ ./Base.jl:419 [inlined]
 [5] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
   @ Base ./loading.jl:1554
in expression starting at /Users/georgegkountouras/.julia/packages/mousetrap/TAzNJ/src/mousetrap.jl:9
in expression starting at stdin:1
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/Pkg/src/Types.jl:67
 [2] precompile(ctx::Pkg.Types.Context, pkgs::Vector{String}; internal_call::Bool, strict::Bool, warn_loaded::Bool, already_instantiated::Bool, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/Pkg/src/API.jl:1435
 [3] precompile
   @ ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/Pkg/src/API.jl:1072 [inlined]
 [4] #precompile#225
   @ ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/Pkg/src/API.jl:1071 [inlined]
 [5] precompile (repeats 2 times)
   @ ~/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/stdlib/v1.8/Pkg/src/API.jl:1071 [inlined]
 [6] top-level scope
   @ REPL[8]:1
```

**Comment by Clemapfel** [comment]
`mousetrap_apple_jll` had an old release for aarch64 that was uploaded under the wrong version number (0.1.1, despite being older than 0.1.0). I replaced it with the up-to-date one, hopefully this fixes it as I don't have any way to test macOS right now

**Comment by ggkountouras** [comment]
The update command was `Pkg.add(url="https://github.com/Clemapfel/mousetrap_apple_jll")`

The issue persists, here's the requested output:

```
The latest version of Julia in the `release` channel is 1.9.2+0.aarch64.apple.darwin14. You currently have `1.8.5+0.aarch64.apple.darwin14` installed. Run:

  juliaup update

to install Julia 1.9.2+0.aarch64.apple.darwin14 and update the `release` channel to that version.
ERROR: ArgumentError: Package mousetrap_apple_jll not found in current path.
- Run `import Pkg; Pkg.add("mousetrap_apple_jll")` to install the mousetrap_apple_jll package.
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1163 [inlined]
 [2] macro expansion
   @ ./lock.jl:223 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1144
error: nm: can't open file: a.out (No such file or directory)
```

**Comment by Clemapfel** [comment]
It says `mousetrap_apple_jll` not found but to install mousetrap that module has to also be installed. Does `using mousetrap_apple_jll` work for you in the REPL? 

The `Pkg.add` just installed that module so I'm unsure as to why it wouldn't be found.

**Comment by ggkountouras** [comment]
> Does `using mousetrap_apple_jll` work for you in the REPL?

Yes, that part works. It's `using mousetrap` that fails.

**Comment by Clemapfel** [comment]
Solved by upgrading to Julia 1.9, I suspect this is because the CxxWrap dependency was too old
